### PR TITLE
List of syringes

### PR DESCRIPTION
**Pull request by Sebosek**

### CODE REVIEW

File: web-jehlomat/src/screens/Nalezy/Components/DarkButton.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿import { styled } from '@mui/system';
+
+const DarkButton = styled('button')({
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/DarkButton.ts:3]: Same as `Button` component. In general; introduce general button component with `component` prop, which one can define underlying element (anchor, button). After that the `DarkButton` should be only the `styled` component with different colors.

File: web-jehlomat/src/screens/Nalezy/Components/ActionLink.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿import { styled } from '@mui/system';
+import { NavLink } from 'react-router-dom';
+
+const ActionLink = styled(NavLink)({
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/ActionLink.ts:4]: Rename to `MenuItem`

File: web-jehlomat/src/screens/Nalezy/Components/Links.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        },
+    },
+});
+
+interface LinksProps {
+    syringe: Syringe;
+    onUpdate: () => void;
+}
+
+const Links: FunctionComponent<LinksProps> = ({ syringe, onUpdate }) => (
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Links.tsx:31]: Rename to `MenuItems`

File: web-jehlomat/src/screens/Nalezy/Components/Map.tsx [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+import { Loader } from 'screens/Nalezy/types/Loader';
+import Loading from 'screens/Nalezy/Components/Loading';
+import PreviewSyringeState from 'screens/Nalezy/Components/SyringeState';
+import Links from 'screens/Nalezy/Components/Links';
+
+import 'leaflet/dist/leaflet.css';
+import gray from 'assets/pins/pin-gray.svg';
+import green from 'assets/pins/pin-green.svg';
+import yellow from 'assets/pins/pin-yellow.svg';
+import dayjs from "dayjs";
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Map.tsx:18]: Formating

File: web-jehlomat/src/screens/Nalezy/Components/Map.tsx [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+const State = styled('div')({
+    alignSelf: 'center',
+    gridColumn: '1 / span 2',
+    gridRow: '2',
+    justifySelf: 'start',
+    padding: '0 8px',
+});
+
+const deriveStateOf = (syringe: Syringe): SyringeState => {
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Map.tsx:73]: Move to utils

File: web-jehlomat/src/screens/Nalezy/Components/Map.tsx [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    }, [error]);
+
+    const coordinates = data.map((item: Syringe) => {
+        const [lat, lng] = item.gps_coordinates.split(',').map(i => i.trim());
+        const coordinate: LatLngTuple = [+lat, +lng];
+
+        return coordinate;
+    });
+
+    const Bounds = (): JSX.Element | null => {
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Map.tsx:118]: Separate to single file component

File: web-jehlomat/src/screens/Nalezy/Components/Map.tsx [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        if (coordinates.length === 0) {
+            return null;
+        }
+
+        const bounds = new L.LatLngBounds(coordinates);
+        map.fitBounds(bounds);
+        return null;
+    };
+
+    const Markers = (): JSX.Element | null => {
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Map.tsx:130]: Separate to single file component

File: web-jehlomat/src/screens/Nalezy/Components/Map.tsx [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+    }
+
+    if (syringe.reservedTill) {
+        return 'RESERVED';
+    }
+
+    return 'WAITING';
+};
+
+const map: Record<SyringeState, Icon> = {
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Map.tsx:85]: Move to utils, possible usage in JH-49

File: web-jehlomat/src/screens/Nalezy/Components/EmptyState.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    width: 300,
+});
+
+const Vertical = styled('div')({
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center',
+
+    '& > *': {
+        marginBottom: 16,
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/EmptyState.tsx:32]: Use `gap` instead

File: web-jehlomat/src/screens/Nalezy/Components/Controls.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
+﻿import { styled } from '@mui/system';
+
+const Controls = styled('div')({
+    display: 'flex',
+
+    '& > *': {
+        marginRight: 12,
```
Comment (Sebosek) [inline web-jehlomat/src/screens/Nalezy/Components/Controls.ts:7]: Use `gap` instead

File: web-jehlomat/src/screens/Nalezy/Components/utils/loadTeams.ts [hunk 11]
```diff
@@ -0,0 +1,17 @@
+﻿import {AxiosResponse} from "axios";
+import {API} from 'config/baseURL';
+
+export interface Team {
+    id: number;
+    name: string;
+}
+
+const loadTeams = async (organizationId: number) => {
+    const url = `/organization/${organizationId}/teams`;
```
Comment (fidelman) [inline web-jehlomat/src/screens/Nalezy/Components/utils/loadTeams.ts:10]: please introduce to apiURL class

File: web-jehlomat/src/screens/Nalezy/Components/utils/loadUsers.ts [hunk 12]
```diff
@@ -0,0 +1,17 @@
+﻿import { AxiosResponse } from 'axios';
+import { API } from 'config/baseURL';
+
+export interface User {
+    id: number;
+    username: string;
+}
+
+const loadUsers = async (organizationId: number) => {
+    const url = `/organization/${organizationId}/users`;
```
Comment (fidelman) [inline web-jehlomat/src/screens/Nalezy/Components/utils/loadUsers.ts:10]: please introduce to apiURL class

**Pull request merged**